app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/TrackListViewModel.kt

```diff
@@ -27,6 +29,31 @@ constructor(
 
   private var _searchQuery = MutableStateFlow("")
 
+  fun toggleTrackSource() {
+    _uiState.value = _uiState.value.copy(showRecentlyAdded = !_uiState.value.showRecentlyAdded)
+    loadTracksBasedOnSource()
+  }
+
+  fun loadTracksBasedOnSource() {
+    viewModelScope.launch {
+      _uiState.value = _uiState.value.copy(loading = true)
+      if (_uiState.value.showRecentlyAdded) {
+        loadRecentlyAddedTracks()
+      } else {
+        _uiState.value = _uiState.value.copy(tracks = listOf(), loading = false)
+      }
+    }
+  }
+
+  fun loadRecentlyAddedTracks() {
+    viewModelScope.launch {
+      appDatabase.trackRecordDao().getAllRecentlyAddedTracks().collect { trackRecords ->
+        val tracks = trackRecords.map { Track(it.trackId, "Unknown Title", "Unknown Artist") }
```

Comment:
What is this "Unknown Title" business? I haven't tested it yet, but this seems like it will just show "Unknown" on all the tracks, no?